nuxeo-jsf/nuxeo-platform-ui-web/src/main/java/org/nuxeo/ecm/platform/ui/web/validator/DocumentConstraintValidator.java

```diff
@@ -217,13 +217,11 @@ protected XPathAndField resolveField(FacesContext context, ValueReference vref,
         } else if (ListItemMapper.class.isAssignableFrom(baseClass)) {
             ListItemMapper mapper = (ListItemMapper) base;
             ProtectedEditableModel model = mapper.getModel();
-            ValueExpression listVe;
-            if (model.getParent() != null) {
-                // move one level up to resolve parent list binding
-                listVe = model.getParent().getBinding();
-            } else {
-                listVe = model.getBinding();
+
+            while (model.getParent() != null) {
+                model = model.getParent();
             }
```

Comment:
i don't get the reasoning here: the mapped model should be the parent model, not the highest list parent